src/modm/platform/uart/stm32/uart.cpp.in

```diff
@@ -244,26 +232,15 @@ MODM_ISR({{ name | upper }})
 %% endif
 {
 	using namespace modm::platform;
-%% if options["buffer.rx"]
-	if ({{ hal }}::isReceiveRegisterNotEmpty()) {
-		// TODO: save the errors
-		uint8_t data;
-		{{ hal }}::read(data);
-		rxBuffer.push(data);
-	}
-%% endif
-%% if options["buffer.tx"]
-	if ({{ hal }}::isTransmitRegisterEmpty()) {
-		if (txBuffer.isEmpty()) {
-			// transmission finished, disable TxEmpty interrupt
-			{{ hal }}::disableInterrupt({{ hal }}::Interrupt::TxEmpty);
-		}
-		else {
-			{{ hal }}::write(txBuffer.get());
-			txBuffer.pop();
-		}
-	}
-%% endif
-	{{ hal }}::acknowledgeInterruptFlags({{ hal }}::InterruptFlag::OverrunError);
+    bool rxResult{false}, txResult{false};
+    if( {{ hal }}::isReceiveRegisterNotEmpty() && {{ hal }}::RxCallback )
+        rxResult = {{ hal }}::RxCallback();
+
+    if( {{ hal }}::isTransmitRegisterEmpty() && {{ hal }}::TxCallback )
+        txResult = {{ hal }}::TxCallback();
+
+    if( {{ hal }}::FinalCallback )
+        {{ hal }}::FinalCallback(rxResult || txResult);
```

Comment:
> Would you be OK with such restriction?

I think that's ok. I don't think it makes sense to mix want to have a TX buffer with a FreeRTOS Queue with a "polling" rx queue.

However, I would like to have the option to have a buffer only for one direction and zero-length buffers. For example, our board logger setup is output only, so it doesn't need a RX buffer at all. Similarly for some UART-based communication protocols like [AMNB](https://modm.io/reference/module/modm-communication-amnb/), it would be a software RX buffer of ≥32B and only the hardware TX buffer (aka software buffer of length 0).